internal/provisioner/kops_provisioner.go

```diff
@@ -715,6 +725,14 @@ func (provisioner *KopsProvisioner) DeleteClusterInstallation(cluster *model.Clu
 		return errors.Wrapf(err, "failed to delete cluster installation %s", clusterInstallation.ID)
 	}
 
+	secretName := fmt.Sprintf("%s-license", makeClusterInstallationName(clusterInstallation))
+	err = k8sClient.DeleteSecret(clusterInstallation.Namespace, secretName)
+	if k8sErrors.IsNotFound(err) {
+		logger.Infof("Secret %s/%s not found. Maybe the license was not set for this installation or was already deleted", clusterInstallation.Namespace, secretName)
```

Comment:
We should be able to handle this better.

If the installation object has `""` for a license value, we should know that there shouldn't be a license file, so we can try to clean it up just in case and then log if we don't find anything like you are doing here.

That said, if the license value is not `""`, we may want to return an error here and not log.

Reply:
updated